src/convert.jl

```diff
@@ -29,21 +165,20 @@ end
     @boundscheck if length(a) != length(SA)
         dimension_mismatch_fail(SA, a)
     end
-
-    return _convert(SA, a, Length(SA))
+    SA′ = construct_type(SA, a)
+    return SA′(unroll_tuple(a, Length(SA′)))
 end
 
-@inline _convert(SA, a, l::Length) = SA(unroll_tuple(a, l))
-@inline _convert(SA::Type{<:StaticArray{<:Tuple,T}}, a, ::Length{0}) where T = similar_type(SA, T)(())
-@inline _convert(SA, a, ::Length{0}) = similar_type(SA, eltype(a))(())
-
 length_val(a::T) where {T <: StaticArrayLike} = length_val(Size(T))
 length_val(a::Type{T}) where {T<:StaticArrayLike} = length_val(Size(T))
 
+unroll_tuple(a::AbstractArray, ::Length{0}) = ()
+unroll_tuple(a::AbstractArray, ::Length{1}) = @inbounds (a[],)
 @generated function unroll_tuple(a::AbstractArray, ::Length{L}) where {L}
-    exprs = [:(a[$j]) for j = 1:L]
+    exprs = (:(a[$j+Δj]) for j = 0:L-1)
     quote
         @_inline_meta
+        Δj = firstindex(a)
```

Comment:
I think this change is a fix for `OffsetArrays.jl`. Could you add tests with the package with the `[extras]` table in `Project.toml`?